### ISSUE TITLE
cli: better error reporting

### ISSUE THREAD

**Issue by anton-khodak** [reanahub/reana-client#43]
Currently, reana-client reports errors in debug mode only and with a very small fraction of the accessible information so additional debugging is necessary in almost all cases to understand what caused the error. The goal is to help a user to diagnose a problem right after the first failure.

Example of current debugging output:
```
/Users/anton/.virtualenvs/reana-py3/bin/python /Users/anton/Programming/NYU-CERN/ak-reana-client/reana_client/cli/cwl_runner.py binding-test.cwl bwa-mem-job.json
/Users/anton/Programming/NYU-CERN/ak-reana-client/reana_client/cli/cwl_runner.py 1.0.20171219150536
INFO:cwltool:/Users/anton/Programming/NYU-CERN/ak-reana-client/reana_client/cli/cwl_runner.py 1.0.20171219150536
Resolved 'binding-test.cwl' to 'file:///Users/anton/Programming/CWL/cwltool/cwltool/schemas/v1.0/v1.0/binding-test.cwl'
INFO:cwltool:Resolved 'binding-test.cwl' to 'file:///Users/anton/Programming/CWL/cwltool/cwltool/schemas/v1.0/v1.0/binding-test.cwl'
DEBUG:past.translation:Running find_module: rdflib.plugins.memory...
DEBUG:past.translation:Running load_module for rdflib.plugins.memory...
DEBUG:past.translation:Excluded rdflib.plugins.memory from translation
DEBUG:past.translation:Running find_module: netrc...
DEBUG:past.translation:Running load_module for netrc...
DEBUG:past.translation:Excluded netrc from translation
DEBUG:past.translation:Running find_module: shlex...
DEBUG:past.translation:Running load_module for shlex...
DEBUG:past.translation:Excluded shlex from translation
Traceback (most recent call last):
  File "/Users/anton/Programming/NYU-CERN/ak-reana-client/reana_client/cli/cwl_runner.py", line 91, in cwl_runner
    reana_spec)
  File "/Users/anton/Programming/NYU-CERN/ak-reana-client/reana_client/api/client.py", line 129, in create_workflow
    raise e
  File "/Users/anton/Programming/NYU-CERN/ak-reana-client/reana_client/api/client.py", line 119, in create_workflow
    reana_spec, sort_keys=True))).result()
  File "/Users/anton/.virtualenvs/reana-py3/lib/python3.6/site-packages/bravado/http_future.py", line 56, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/anton/.virtualenvs/reana-py3/lib/python3.6/site-packages/bravado/http_future.py", line 115, in result
    self.response_callbacks)
  File "/Users/anton/.virtualenvs/reana-py3/lib/python3.6/site-packages/bravado/http_future.py", line 149, in unmarshal_response
    raise_on_unexpected(incoming_response)
  File "/Users/anton/.virtualenvs/reana-py3/lib/python3.6/site-packages/bravado/http_future.py", line 178, in raise_on_unexpected
    raise make_http_exception(response=http_response)
bravado.exception.HTTPInternalServerError: 500 INTERNAL SERVER ERROR
ERROR:root:None
ERROR:root:500 INTERNAL SERVER ERROR
```

**Comment by tiborsimko** [comment]
Better traceback propagation is now being discussed for the `Internal-Consolidation` sprint: https://github.com/reanahub/reana/issues/55

**Comment by tiborsimko** [comment]
This has been implemented in REANA 0.3.0.
